packages/matter.js/src/device/PairedNode.ts

```diff
@@ -287,106 +392,144 @@ export class PairedNode {
             }
         }
 
-        while (true) {
-            this.setConnectionState(NodeStateInformation.WaitingForDeviceDiscovery);
-
-            if (this.interactionClient !== undefined) {
-                this.interactionClient.close();
-                this.interactionClient = undefined;
-            }
+        this.setConnectionState(NodeStateInformation.WaitingForDeviceDiscovery);
 
-            try {
-                await this.initialize();
-                this.reconnectionInProgress = false;
-                return;
-            } catch (error) {
-                MatterError.accept(error);
+        try {
+            await this.initialize();
+        } catch (error) {
+            MatterError.accept(error);
 
-                if (
-                    this.connectionState === NodeStateInformation.Disconnected ||
-                    this.connectionState === NodeStateInformation.Decommissioned
-                ) {
-                    this.reconnectionInProgress = false;
-                    logger.info("No reconnection desired because requested status is Disconnected.");
-                    return;
-                }
+            if (error instanceof UnknownNodeError) {
+                logger.info(`Node ${this.nodeId}: Device not found, consider decommissioned.`);
```

Comment:
Is this message correct?  We set state to disconnected rather than decommissioned

Reply:
I adjusted the text. And also added an decommissioned event because was missing